lib/publiccloud/google.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Google Cloud Platform
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+#             Jose Lausuch <jalausuch@suse.de>
+
+package publiccloud::google;
+use Mojo::Base 'publiccloud::provider';
+use testapi;
+use strict;
+use utils;
+
+sub init {
+    my ($self) = @_;
+    assert_script_run("curl " . data_url('publiccloud/googlekey.json') . " -o " . $self->key_file);
+    assert_script_run('source ~/.bashrc');
+    assert_script_run('ntpdate -s time.google.com');
+    assert_script_run('gcloud config set account ' . $self->account_name);
+    assert_script_run('gcloud auth activate-service-account ' . $self->service_account . ' --key-file=' . $self->key_file . ' --project=' . $self->project);
+}
+
+sub file2name {
+    my ($self, $file) = @_;
+    my $name = $file;
+    $name = lc $file;           # lower case
+    $name =~ s/[^a-z0-9]//g;    # only allowed characteres from Google Cloud
+    $name =~ s/targz//g;        # removes targz
+    return $name;
+}
+
+sub find_img {
+    my ($self, $name) = @_;
+    my $img_name = $self->file2name($name);
+    my $out      = script_output("gcloud compute images list --filter='name~$img_name' |grep -v NAME|grep -v Listed");
+    return $out;
```

Comment:
That find_img() method should only return something, if the image exists on CSP.